prow/resultstore/uploader.go

```diff
@@ -58,9 +62,54 @@ func (u *Uploader) Upload(ctx context.Context, log *logrus.Entry, p *Payload) er
 	w.WriteTarget(ctx, p.OverallTarget())
 	w.WriteConfiguredTarget(ctx, p.ConfiguredTarget())
 	w.WriteAction(ctx, p.OverallAction())
-	// TODO: When writer.New() is emended to resume uploads (see TODO there),
-	// return non-permanent errors from Finalize(). Doing this now is futile,
-	// since attempting to upload an existing invocation is a permanent error.
-	w.Finalize(ctx)
+
+	if err := w.Finalize(ctx); err != nil {
+		if writer.PermanentError(err) {
+			return nil
+		}
+		return err
+	}
 	return nil
 }
+
+type tokener struct {
```

Comment:
Nit: The word "token" is not itself a verb so "tokener" has a touch of mystery about it. Why not something like tokenCreator, tokenFactory, or similar?

Reply:
Changed to tokenGenerator.